OVERVIEW.md

```diff
@@ -0,0 +1,484 @@
+
+```
+ocr -> PluginThatImplementsLibOCR2
+filter -> ParallelServiceWatchingUpkeepLogs
+
+call ocr.ShouldTransmitAcceptedReport with [ReportTimestamp] and [ReportBytes]
+    call decode with (ReportBytes): return [upkeepStateResultList]
+    if err in decode
+        return false, and error
+
+    if length of (upkeepStateResultList) == 0
+        return false, and error
+
+    for each (upkeepStateResultList)
+        call filter.IsTransmissionConfirmed with [upkeepStateResult]
+        if upkeep is not confirmed
+            return true, and no error
+
+    return false, and no error
+```
+
+## Parallel Services
+
+### Poller
+
+This service interfaces with the registry contract and pulls a list of active
+upkeeps for every block. It is up to the registry to maintain the list of 
+active upkeeps either by itself polling the contract or by listening for logs.
+
+**Sampling Ratio**
+
+The sampling ratio is defined as the ratio of upkeeps a single node MUST check
+at random such that `n` good nodes have `p` probability of checking all upkeeps 
+at least once over `r` range of blocks.
+
+This limits the workload of individual nodes and attempts to distribute the 
+total workload equally to all nodes while synchronizing only on `n`, `p`, 
+and `r`.
+
+Faulty node limit `f` is included in the calculation of the sampling ratio.
+
+**Polling**
+
+For each block, this service attempts to check the state of all active upkeeps
+provided by the registry. First, all upkeeps are shuffled before checking
+their state. This ensures that all upkeeps have a chance at being checked over a
+series of blocks. Second, the list of active upkeeps is reduced to the sampling
+size limit.
+
+Each upkeep is then checked at the provided block number and the total results
+are used to reset an internal cache such that the results for the latest block
+replace the results for the previous block.
+
+```
+contract -> RegistryContractInterface
+
+for each [Block]
+    define [activeUpkeeps]
+    for each [upkeep]
+        if upkeep is active and not cancelled
+            add (upkeep) to (activeUpkeeps)
+
+    call shuffle with (activeUpkeeps)
+    replace (activeUpkeeps) with slice of (activeUpkeeps:sampleRatioLimit)
+
+    define [upkeepResults]
+    for each (activeUpkeeps)
+        call contract.CheckUpkeep with [upkeep] and (Block): return [upkeepResult]
+        if no error
+            add (upkeepResult) to (upkeepResults)
+
+    replace (poller.upkeepResultsCache) with (upkeepResults)
+```
+
+**Providing Samples**
+
+Samples are provided directly from the cached results of the last poll. All
+'in-flight' results are filtered out at this point.
+
+```
+if no results sampled
+    return no results, and error
+
+define [blockNumber]
+define [upkeepResults]
+for each (upkeepResults)
+    if [upkeepResult] is Pending
+        remove (upkeepResult) from (upkeepResults)
+
+return (blockNumber), (upkeepResults), and no error
+```
+
+### Filter
```

Comment:
This section is obviously quite important, but not very clearly written, at least from my perspective. I suggest careful review and reformulation of the design decisions here, as errors and oversights may be hiding in the unclear parts.